bonfire/Modules/Tools/Controllers/LogsController.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace Bonfire\Tools\Controllers;
+
+use App\Controllers\AdminController;
+use CodeIgniter\CodeIgniter;
+
+class LogsController extends AdminController
+{
+    protected $theme = 'Admin';
+
+    protected $viewPrefix = 'Bonfire\Modules\Tools\Views\\';
+
+    protected $logs_path = WRITEPATH . 'logs/';
+
+    private static $levelsIcon = [
+        'INFO'  => 'fas fa-info-circle',
+        'ERROR' => 'fas fa-times',
+        'CRITICAL' => 'fas fa-exclamation-triangle',
+        'DEBUG' => 'fas fa-bug',
+        'ALL'   => 'fas fa-minus',
+    ];
+
+    private static $levelClasses = [
+        'INFO'     => 'info',
+        'ERROR'    => 'danger',
+        'CRITICAL' => 'danger',
+        'DEBUG'    => 'warning',
+        'ALL'      => 'muted',
+    ];
+
+    const MAX_LOG_SIZE = 52428800; //50MB
+    const MAX_STRING_LENGTH = 350; //300 chars
+
+    const LOG_LINE_START_PATTERN = "/((INFO)|(ERROR)|(CRITICAL)|(DEBUG)|(ALL))[\s\-\d:\.\/]+(-->)/";
+    const LOG_DATE_PATTERN = ["/^((ERROR)|(CRITICAL)|(INFO)|(DEBUG)|(ALL))\s\-\s/", "/\s(-->)/"];
+    const LOG_LEVEL_PATTERN = "/^((ERROR)|(CRITICAL)|(INFO)|(DEBUG)|(ALL))/";
+
+    /**
+     * Displays all logs.
+     *
+     * @return string
+     */
+    public function index()
+    {
+        
+      if (isset($_POST['delete'])) {
```

Comment:
Please move the handling of the `delete` and `delete_all` actions to a new method. If the purpose of the `index` method is to display all logs it should be limited to that functionality.